fetcher.go

```diff
@@ -16,10 +16,16 @@ import (
 	"github.com/ipld/go-ipld-prime/traversal/selector/builder"
 )
 
-// TODO: need to support sessions
+type FetcherConfig struct {
+	blockService blockservice.BlockService
+}
 
 type Fetcher struct {
-	exchange *bitswap.Bitswap
+	// TODO: for now, passing this to instantiation of block session is enough to
+	// cancel on session context cancel, but we may want to use this direct reference
+	// more tightly in this code
+	ctx         context.Context
```

Comment:
This is private and unused. I get why we might want it some day, but it will be easy to add, so we should drop it for now.

Reply:
fair